// @aws-sdk/client-rekognition/dist-es/commands/GetFaceSearchCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetFaceSearchRequest, GetFaceSearchResponse } from "../models/models_0";
import { deserializeAws_json1_1GetFaceSearchCommand, serializeAws_json1_1GetFaceSearchCommand, } from "../protocols/Aws_json1_1";
var GetFaceSearchCommand = (function (_super) {
    __extends(GetFaceSearchCommand, _super);
    function GetFaceSearchCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetFaceSearchCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RekognitionClient";
        var commandName = "GetFaceSearchCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetFaceSearchRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetFaceSearchResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetFaceSearchCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetFaceSearchCommand(input, context);
    };
    GetFaceSearchCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetFaceSearchCommand(output, context);
    };
    return GetFaceSearchCommand;
}($Command));
export { GetFaceSearchCommand };
