// @aws-sdk/client-rekognition/dist-es/commands/GetSegmentDetectionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetSegmentDetectionRequest, GetSegmentDetectionResponse } from "../models/models_0";
import { deserializeAws_json1_1GetSegmentDetectionCommand, serializeAws_json1_1GetSegmentDetectionCommand, } from "../protocols/Aws_json1_1";
var GetSegmentDetectionCommand = (function (_super) {
    __extends(GetSegmentDetectionCommand, _super);
    function GetSegmentDetectionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetSegmentDetectionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RekognitionClient";
        var commandName = "GetSegmentDetectionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetSegmentDetectionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetSegmentDetectionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetSegmentDetectionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetSegmentDetectionCommand(input, context);
    };
    GetSegmentDetectionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetSegmentDetectionCommand(output, context);
    };
    return GetSegmentDetectionCommand;
}($Command));
export { GetSegmentDetectionCommand };
