// @aws-sdk/client-rekognition/dist-es/commands/IndexFacesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { IndexFacesRequest, IndexFacesResponse } from "../models/models_0";
import { deserializeAws_json1_1IndexFacesCommand, serializeAws_json1_1IndexFacesCommand, } from "../protocols/Aws_json1_1";
var IndexFacesCommand = (function (_super) {
    __extends(IndexFacesCommand, _super);
    function IndexFacesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    IndexFacesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RekognitionClient";
        var commandName = "IndexFacesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: IndexFacesRequest.filterSensitiveLog,
            outputFilterSensitiveLog: IndexFacesResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    IndexFacesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1IndexFacesCommand(input, context);
    };
    IndexFacesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1IndexFacesCommand(output, context);
    };
    return IndexFacesCommand;
}($Command));
export { IndexFacesCommand };
