// @aws-sdk/client-rekognition/dist-es/commands/GetCelebrityRecognitionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetCelebrityRecognitionRequest, GetCelebrityRecognitionResponse } from "../models/models_0";
import { deserializeAws_json1_1GetCelebrityRecognitionCommand, serializeAws_json1_1GetCelebrityRecognitionCommand, } from "../protocols/Aws_json1_1";
var GetCelebrityRecognitionCommand = (function (_super) {
    __extends(GetCelebrityRecognitionCommand, _super);
    function GetCelebrityRecognitionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetCelebrityRecognitionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RekognitionClient";
        var commandName = "GetCelebrityRecognitionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetCelebrityRecognitionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetCelebrityRecognitionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetCelebrityRecognitionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetCelebrityRecognitionCommand(input, context);
    };
    GetCelebrityRecognitionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetCelebrityRecognitionCommand(output, context);
    };
    return GetCelebrityRecognitionCommand;
}($Command));
export { GetCelebrityRecognitionCommand };
