// @aws-sdk/client-rekognition/dist-es/commands/GetLabelDetectionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetLabelDetectionRequest, GetLabelDetectionResponse } from "../models/models_0";
import { deserializeAws_json1_1GetLabelDetectionCommand, serializeAws_json1_1GetLabelDetectionCommand, } from "../protocols/Aws_json1_1";
var GetLabelDetectionCommand = (function (_super) {
    __extends(GetLabelDetectionCommand, _super);
    function GetLabelDetectionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetLabelDetectionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "RekognitionClient";
        var commandName = "GetLabelDetectionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetLabelDetectionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: GetLabelDetectionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetLabelDetectionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetLabelDetectionCommand(input, context);
    };
    GetLabelDetectionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetLabelDetectionCommand(output, context);
    };
    return GetLabelDetectionCommand;
}($Command));
export { GetLabelDetectionCommand };
